### PR TITLE
Implement demo profiles

### PR DESCRIPTION
**Pull request by vopswtf**
should work fine, entitlements check was randomly unauthorized sometimes but I think thats because of the current outage.

### CODE REVIEW

**Comment by vopswtf** [comment]
left log4j debug in the xml 🙏

File: launcher-patches/0031-Implement-demo-profiles.patch [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+diff --git a/src/main/resources/log4j2.xml b/src/main/resources/log4j2.xml
+index 720caf7..0e7c4dd 100644
+--- a/src/main/resources/log4j2.xml
++++ b/src/main/resources/log4j2.xml
+@@ -13,7 +13,7 @@
+         </Async>
+     </Appenders>
+     <Loggers>
+-        <Root level="info">
++        <Root level="debug">
```
Comment (bigfoot547) [inline launcher-patches/0031-Implement-demo-profiles.patch:129]: leaving this here to remind me to check this before merging

File: launcher-patches/0031-Implement-demo-profiles.patch [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
++                } catch (InterruptedException | InvocationTargetException e) {
++                    throw new RuntimeException(e);
++                }
++                profileInfo = getAuthenticationService().getAuthenticated(MicrosoftAuthenticationService.ROUTE_MINECRAFT_PROFILE,
++                        mojToken, MCProfileInfo.class);
+             }
+-            profileInfo = getAuthenticationService().getAuthenticated(MicrosoftAuthenticationService.ROUTE_MINECRAFT_PROFILE,
+-                    mojToken, MCProfileInfo.class);
+-        }
+         }
```
Comment (bigfoot547) [inline launcher-patches/0031-Implement-demo-profiles.patch:90]: you could reduce clutter in the diff if you just don't reindent this

**Comment by vopswtf** [comment]
fixed clutter + log4j

**Comment by bigfoot547** [comment]
note: if your account has not been set up on xbox, you will get this error:
```
com.mojang.authlib.exceptions.AuthenticationException: Account is not on Xbox. 2148916233 ()
	at dev.figboot.olauncher.auth.MicrosoftUserAuthentication.xstsAndMinecraftLogIn(MicrosoftUserAuthentication.java:235) ~[patched.jar:1.7.3_01]
	at dev.figboot.olauncher.auth.MicrosoftUserAuthentication.logIn(MicrosoftUserAuthentication.java:130) ~[patched.jar:1.7.3_01]
	at net.minecraft.launcher.ui.popups.login.LogInForm$2.run(LogInForm.java:165) ~[patched.jar:1.7.3_01]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
	at java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: dev.figboot.olauncher.auth.error.ResponseCodeException: java.io.IOException: Server returned HTTP response code: 401 for URL: https://xsts.auth.xboxlive.com/xsts/authorize
	at dev.figboot.olauncher.auth.MicrosoftAuthenticationService.readResponse(MicrosoftAuthenticationService.java:111) ~[patched.jar:1.7.3_01]
	at dev.figboot.olauncher.auth.MicrosoftAuthenticationService.makeRequest(MicrosoftAuthenticationService.java:144) ~[patched.jar:1.7.3_01]
	at dev.figboot.olauncher.auth.MicrosoftAuthenticationService.makeRequest(MicrosoftAuthenticationService.java:118) ~[patched.jar:1.7.3_01]
	at dev.figboot.olauncher.auth.MicrosoftAuthenticationService.post(MicrosoftAuthenticationService.java:153) ~[patched.jar:1.7.3_01]
	at dev.figboot.olauncher.auth.MicrosoftUserAuthentication.xstsAndMinecraftLogIn(MicrosoftUserAuthentication.java:224) ~[patched.jar:1.7.3_01]
	... 5 more
Caused by: java.io.IOException: Server returned HTTP response code: 401 for URL: https://xsts.auth.xboxlive.com/xsts/authorize
	at jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62) ~[?:?]
	at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502) ~[?:?]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:486) ~[?:?]
	at sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:2055) ~[?:?]
	at sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:2050) ~[?:?]
	at java.security.AccessController.doPrivileged(AccessController.java:571) ~[?:?]
	at sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:2049) ~[?:?]
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1619) ~[?:?]
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1599) ~[?:?]
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:223) ~[?:?]
	at dev.figboot.olauncher.auth.MicrosoftAuthenticationService.readResponse(MicrosoftAuthenticationService.java:95) ~[patched.jar:1.7.3_01]
	at dev.figboot.olauncher.auth.MicrosoftAuthenticationService.makeRequest(MicrosoftAuthenticationService.java:144) ~[patched.jar:1.7.3_01]
	at dev.figboot.olauncher.auth.MicrosoftAuthenticationService.makeRequest(MicrosoftAuthenticationService.java:118) ~[patched.jar:1.7.3_01]
	at dev.figboot.olauncher.auth.MicrosoftAuthenticationService.post(MicrosoftAuthenticationService.java:153) ~[patched.jar:1.7.3_01]
	at dev.figboot.olauncher.auth.MicrosoftUserAuthentication.xstsAndMinecraftLogIn(MicrosoftUserAuthentication.java:224) ~[patched.jar:1.7.3_01]
	... 5 more
Caused by: java.io.IOException: Server returned HTTP response code: 401 for URL: https://xsts.auth.xboxlive.com/xsts/authorize
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1998) ~[?:?]
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1599) ~[?:?]
	at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:531) ~[?:?]
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:307) ~[?:?]
	at dev.figboot.olauncher.auth.MicrosoftAuthenticationService.readResponse(MicrosoftAuthenticationService.java:90) ~[patched.jar:1.7.3_01]
	at dev.figboot.olauncher.auth.MicrosoftAuthenticationService.makeRequest(MicrosoftAuthenticationService.java:144) ~[patched.jar:1.7.3_01]
	at dev.figboot.olauncher.auth.MicrosoftAuthenticationService.makeRequest(MicrosoftAuthenticationService.java:118) ~[patched.jar:1.7.3_01]
	at dev.figboot.olauncher.auth.MicrosoftAuthenticationService.post(MicrosoftAuthenticationService.java:153) ~[patched.jar:1.7.3_01]
	at dev.figboot.olauncher.auth.MicrosoftUserAuthentication.xstsAndMinecraftLogIn(MicrosoftUserAuthentication.java:224) ~[patched.jar:1.7.3_01]
	... 5 more
```

**Review by bigfoot547 — APPROVED** [review]
just tested, it works. let me know if the above thing is intentional (it seems so - if the user is not enrolled in xboxlive, they will not have an xuid which is required). if so, I will merge it.

**Comment by vopswtf** [comment]
does it return to the login screen? or does it infinite load?

i feel like there shouldnt be any instances of infinite loading at the login screen atleast

**Comment by bigfoot547** [comment]
it returns to the login screen reporting a generic 401 response

**Comment by vopswtf** [comment]
that should be fine for merge then

**Comment by bigfoot547** [comment]
74eeae330ffd924a3843a5dab76f837ac01a3bfb

**Pull request closed without merging**